src/apps/cli/SandboxSkill.cs

```diff
@@ -1,38 +1,49 @@
+using System.Text;
 using DotNet.Testcontainers.Builders;
-using Microsoft.SemanticKernel.SkillDefinition;
+using DotNet.Testcontainers.Configurations;
+using DotNet.Testcontainers.Containers;
 
 public class SandboxSkill
 {
-    public async Task<string> RunInAlpineAsync(string input)
+    public Task<string> RunInAlpineAsync(string input)
     {
-        return await RunInContainer(input, "alpine");
+        return RunInContainer(input, "alpine:3.18");
     }
 
-    public async Task<string> RunInDotnetAlpineAsync(string input)
+    public Task<string> RunInDotnetAlpineAsync(string input)
     {
-        return await RunInContainer(input, "mcr.microsoft.com/dotnet/sdk:7.0");
+        return RunInContainer(input, "mcr.microsoft.com/dotnet/sdk:7.0-alpine");
     }
 
-    private async Task<string> RunInContainer(string input, string image)
+    private static async Task<string> RunInContainer(string input, string image)
     {
-        var tempScriptFile = $"{Guid.NewGuid().ToString()}.sh";
-        var tempScriptPath = $"./output/{tempScriptFile}";
-        await File.WriteAllTextAsync(tempScriptPath, input);
-        Directory.CreateDirectory(Path.Combine(Directory.GetCurrentDirectory(),"output", "src"));
+        var tempScriptFile = Path.ChangeExtension(Guid.NewGuid().ToString(), "sh");
+
+        var srcDirectoryPath = Path.Combine(Directory.GetCurrentDirectory(), "output", "src");
+        _ = Directory.CreateDirectory(srcDirectoryPath);
+
         var dotnetContainer = new ContainerBuilder()
-                            .WithName(Guid.NewGuid().ToString("D"))
-                            .WithImage(image)
-                            .WithBindMount(Path.Combine(Directory.GetCurrentDirectory(),"output", "src"), "/src")
-                            .WithBindMount(Path.Combine(Directory.GetCurrentDirectory(), tempScriptPath), $"/src/{tempScriptFile}")
-                            .WithWorkingDirectory("/src")
-                            .WithCommand("sh", tempScriptFile)
-                            .Build();
+            .WithImage(image)
+            .WithBindMount(srcDirectoryPath, "/src")
+            .WithResourceMapping(Encoding.Default.GetBytes(input), $"/src/{tempScriptFile}")
+            .WithWaitStrategy(Wait.ForUnixContainer().AddCustomWaitStrategy(new ScriptCompleted()))
```

Comment:
The wait strategy I have implemented assumes that the container's startup process should wait until the shell script has completed.